File: kthSymbolInGrammar/kthSymbolInGrammar.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int kthGrammar(int N, int K) {
        if (N == 1 && K == 1)
            return 0;
        if (K % 2 == 0)
            return kthGrammar(N - 1, K / 2) == 0 ? 1 : 0;
        else 
            return kthGrammar(N - 1, (K + 1) / 2) == 0 ? 0 : 1;
    }
};

int main() {
    Solution s;
    cout << s.kthGrammar(5, 10) << endl;
    return 0;
}

File: 779.kthSymbolInGrammar/kthSymbolInGrammar.java
class Solution {
    public int kthGrammar(int n, int k) {
        if (n == 1 && k == 1) return 0;
        if (k % 2 != 0) { 
            return kthGrammar(n - 1, (k - 1) / 2 + 1);
        } else {
            if (kthGrammar(n - 1, (k - 1) / 2 + 1) == 1) return 0;
            else return 1;
        }
    }
}
